functions/src/models/offers/index.ts

```diff
@@ -149,9 +150,17 @@ export class Offer implements IOffer {
   }
 }
 
-export const OfferFirestoreConverter: firebase.firestore.FirestoreDataConverter<Offer> = {
+export const OfferFirestoreConverter: FirestoreDataConverter<Offer> = {
   fromFirestore: (data: QueryDocumentSnapshot<IOffer>): Offer => {
     return Offer.factory(data.data());
   },
-  toFirestore: (modelObject: Offer): DocumentData => modelObject.toObject(),
+  toFirestore: (modelObject: Offer): DocumentData => ({
+    cavUserRef: modelObject.cavUserRef,
+    pinUserRef: modelObject.pinUserRef,
+    requestRef: modelObject.requestRef,
+    cavUserSnapshot: modelObject.cavUserSnapshot.toObject(),
+    message: modelObject.message,
+    status: modelObject.status,
+    createdAt: modelObject.createdAt.toDate(),
+  }),
```

Comment:
Is there any reason we moved away from `model.toObject` here? We don't want to have to maintain this code in two functions.

Reply:
yea, this was because we needed to convert the timestamps into Date objects for the tests to pass.
But when I tried the same on the response that I sent to the client, it was behaving strangely (the dates weren't having any values for some reason). I wasn't able to figure out why.
So when it threw an error with the timestamp in the tests, I thought of not making the change in the `.toObject` the method as that's the one we are using in the cloud function and aren't using the converter for the reason I mentioned to you regarding the errors I was getting with the converters in the cloud function and you were using the same method too